dkg/frostp2p.go

```diff
@@ -70,7 +76,45 @@ func newFrostP2P(tcpNode host.Host, peers map[peer.ID]cluster.NodeIdx, bcastComp
 		round1CastsRecv: round1CastsRecv,
 		round1P2PRecv:   round1P2PRecv,
 		round2CastsRecv: round2CastsRecv,
+	}, nil
+}
+
+// newCheckMsg returns a bcast.CheckMessage function for round 1 and 2.
+func newCheckMsg(messageID string) (bcast.CheckMessage, error) {
+	found := false
+	for _, mID := range frostMessageIDs() {
+		if mID == messageID {
+			found = true
+		}
 	}
+
+	if !found {
+		return nil, errors.New("frost message id unsupported", z.Str("message_id", messageID))
+	}
+
+	return func(ctx context.Context, peerID peer.ID, msgAny *anypb.Any) error {
+		targetFn := func(messageID string) proto.Message {
+			switch messageID {
+			case round1CastID:
+				r := pb.FrostRound1Casts{}
+				return &r
+			case round2CastID:
+				r := pb.FrostRound2Casts{}
+				return &r
+			default:
+				return nil // Note: impossible due to the above check, but needed because compilers.
+			}
+		}
+
+		target := targetFn(messageID)
+
+		err := msgAny.UnmarshalTo(target)
```

Comment:
this is still an incomplete solution as a malicious peer can still broadcast a malformed message in the correct `pb.FrostRount1Casts`. They can still be able to get the sign on this data from other peers.

Reply:
They don't control the input anymore though, i.e. it's not just a slice of bytes.

What do you suggest we check?

Reply:
not sure if we can check each field in `pb.FrostRound1Cast`